modules/mt/keeper/keeper.go

```diff
@@ -130,9 +152,18 @@ func (k Keeper) BurnMT(ctx sdk.Context, denomID, tokenID string, owner sdk.AccAd
 		return err
 	}
 
-	k.deleteMT(ctx, denomID, mt)
-	k.deleteOwner(ctx, denomID, tokenID, owner)
-	k.decreaseSupply(ctx, denomID)
+	srcOwnerAmount := k.getOwner(ctx, denomID, tokenID, owner)
+	if srcOwnerAmount < amount {
+		return sdkerrors.Wrapf(types.ErrInvalidCollection, "Lack of mt: %", srcOwnerAmount)
+	}
+
+	k.deleteOwner(ctx, denomID, tokenID, amount, owner)
+	k.setMT(ctx, denomID, types.MT{
```

Comment:
There is a `increaseSupply` above, so this should be done in the `decreaseSupply` method